trin-core/src/portalnet/overlay_service.rs

```diff
@@ -309,6 +316,8 @@ impl<
     > OverlayService<TContentKey, TMetric, TValidator>
 where
     <TContentKey as TryFrom<Vec<u8>>>::Error: Debug,
+    TValidator: Sync,
+    TMetric: Sync,
```

Comment:
nit: Why not add these bounds in the `impl<..>` above where we already bound with `Send`?